### DISCUSSION TITLE
Multi inverters

### DISCUSSION THREAD

**Discussion by endurasek** [macxq/foxess-ha#55] [Q&A] [answered]
Hello
I have a problem with reading 2 inverters
I can always read data from only one, but never from two at the same time

# Loads default set of integrations. Do not remove.
default_config:

# Text to speech
tts:
  - platform: google_translate

automation: !include automations.yaml
script: !include scripts.yaml
scene: !include scenes.yaml

sensor:
  - platform: foxess
    name: FoxESS1
    username: login
    password: Haslo
    deviceID: 052...
  - platform: foxess
    name: FoxESS2
    username: login
    password: Haslo
    deviceID: c19...
    
logger:
  default: warning
  logs:
    custom_components.foxess: debug

**Comment by macxq** [comment]
Can you provide logs?

https://github.com/macxq/foxess-ha#-troubleshooting

**Comment by endurasek** [comment]
tauron_amiplus: Error on device update!
19:08:19 – (BŁĘDY) Sensor (niestandardowa integracja)
Setup of sensor platform tauron_amiplus is taking over 10 seconds.
19:07:48 – (OSTRZEŻENIA) Sensor
FoxESS Cloud initializaction failed, fix error and restar ha
19:07:40 – (BŁĘDY) foxess (niestandardowa integracja)
Unexpected error fetching FoxESS data: 'NoneType' object is not iterable
19:07:40 – (BŁĘDY) foxess (niestandardowa integracja)
Ended unfinished session (id=24 from 2022-04-18 15:09:24.788400)
19:07:31 – (OSTRZEŻENIA) Recorder
The system could not validate that the sqlite3 database at //config/home-assistant_v2.db was shutdown cleanly
19:07:31 – (OSTRZEŻENIA) Recorder

**Comment by macxq** [comment]
which version are you using?

**Comment by endurasek** [comment]
v0.15-beta2

**Comment by endurasek** [comment]
![Screenshot_2022-04-18-19-54-57-25_40deb401b9ffe8e1df2f1cc5ba480b12](https://user-images.githubusercontent.com/103924330/163851824-97f3e827-6d03-4b24-9eff-b67745a71b42.jpg)

**Comment by macxq** [comment]
when you configure it separately (just one device at once) does it work ?

**Reply by endurasek** [reply]
if I add two, only one will work.  if I add one it will work one.  i am using the same configuration for two inverters.  only if I want to turn off one, I use #

**Comment by williamjeccles** [comment, marked as answer]
Interesting my dual inverter setup is working fine.

**Comment by endurasek** [comment]
I don't know why but only one inverter can read data at a time
![fox](https://user-images.githubusercontent.com/103924330/174027359-7d83a2a9-4728-43f6-9e54-2015da9976ac.jpg)
